homeassistant/components/smlight/sensor.py

```diff
@@ -0,0 +1,163 @@
+"""Support for SLZB-06 sensors."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+
+from pysmlight.web import Sensors
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import EntityCategory, UnitOfInformation, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.dt import utcnow
+
+from . import SmConfigEntry
+from .const import LOGGER, SCAN_INTERVAL, SMLIGHT_SLZB_REBOOT_EVENT, UPTIME_DEVIATION
+from .coordinator import SmDataUpdateCoordinator
+from .entity import SmEntity
+
+
+@dataclass(frozen=True, kw_only=True)
+class SmSensorEntityDescription(SensorEntityDescription):
+    """Class describing SMLIGHT sensor entities."""
+
+    entity_category: EntityCategory = EntityCategory.DIAGNOSTIC
+    value_fn: Callable[[Sensors], float | None]
+
+
+SENSORS = [
+    SmSensorEntityDescription(
+        key="core_temperature",
+        translation_key="core_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        value_fn=lambda x: x.esp32_temp,
+    ),
+    SmSensorEntityDescription(
+        key="zigbee_temperature",
+        translation_key="zigbee_temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+        value_fn=lambda x: x.zb_temp,
+    ),
+    SmSensorEntityDescription(
+        key="ram_usage",
+        translation_key="ram_usage",
+        device_class=SensorDeviceClass.DATA_SIZE,
+        native_unit_of_measurement=UnitOfInformation.KILOBYTES,
+        entity_registry_enabled_default=False,
+        value_fn=lambda x: x.ram_usage,
+    ),
+    SmSensorEntityDescription(
+        key="fs_usage",
+        translation_key="fs_usage",
+        device_class=SensorDeviceClass.DATA_SIZE,
+        native_unit_of_measurement=UnitOfInformation.KILOBYTES,
+        entity_registry_enabled_default=False,
+        value_fn=lambda x: x.fs_used,
+    ),
+    SmSensorEntityDescription(
+        key="core_uptime",
+        translation_key="core_uptime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_registry_enabled_default=False,
+        value_fn=lambda x: x.uptime,
+    ),
+    SmSensorEntityDescription(
+        key="socket_uptime",
+        translation_key="socket_uptime",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        entity_registry_enabled_default=False,
+        value_fn=lambda x: x.socket_uptime if x.socket_uptime > 0 else None,
```

Comment:
What do theser eturn?

Reply:
socket uptime is reported as 0 when it is disconnected, but we want this to display as `Unavailable`. I probably should move this to the library. removed now, will fix for uptime PR